src/instruction_generator/instruction_generator.rs

```diff
@@ -211,9 +210,76 @@ impl InstructionGenerator {
         );
     }
 
-    pub fn generate_assignment_instructions(&mut self, l: QNameNode, r: ExpressionNode) {
-        self.generate_expression_instructions(r);
-        let Locatable { element, pos } = l;
-        self.push(Instruction::Store(element), pos);
+    pub fn generate_assignment_instructions(
+        &mut self,
+        l: ResolvedDeclaredName,
+        r: ExpressionNode,
+        pos: Location,
+    ) {
+        let left_type = l.type_definition();
+        self.generate_expression_instructions_casting(r, left_type);
+        self.push(Instruction::Store(l), pos);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::common::{AtRowCol, StripLocation};
+    use crate::instruction_generator::{generate_instructions, Instruction, InstructionNode};
+    use crate::linter::{lint, ResolvedDeclaredName};
+    use crate::parser::{parse_main_str, TypeQualifier};
+    use crate::variant::Variant;
+
+    pub fn generate_instructions_str<T>(input: T) -> Vec<InstructionNode>
```

Comment:
move to new `test_utils` of instruction generator, reuse in interpreter test_utils